ui/src/components/MessageSnackBar.vue

```diff
@@ -0,0 +1,96 @@
+<template>
+  <!-- display a message or error to the end user
+    usage: <MessageSnackBar v-bind:bus="bus"></MessageSnackBar>
+      where bus is a vue instance.
+    do bus.$emit('message', 'message to be displayed')
+    can replace 'message' with 'error' or 'notification' for different style of messages
+    for reference of the event bus model, see https://code.luasoftware.com/tutorials/vuejs/parent-call-child-component-method/
+    -->
+  <div>
+    <!-- TODO: add multiple messages support? 
+      Currently new messages will override the old ones, and count down will not restart <2020-06-04, David Deng> -->
+    <!-- START: to be removed -->
+    <v-btn
+      color="primary"
+      :text="true"
+      @click="$http.get('api/v1/groups/group-types/1')"
+      >make an invalid request</v-btn
+    >
+    <!-- END: to be removed -->
+    <v-snackbar
+      v-model="snackBarObj.show"
+      v-bind:timeout="snackBarObj.timeout"
+      v-bind:color="snackBarObj.color"
+      multi-line
+      top
+    >
+      <blog-post v-if= "snackBarObj.content == '404'">{{ $t("events.error-result-not-found") }}</blog-post>
```

Comment:
What is a `blog-post` control? Is this a new Vuetify thing?

Reply:
I will review these code again... Maybe close this PR for now, and we will make a new one when it is more ready (I just wanted some of your feedback on the logging functionality).